src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -802,17 +802,52 @@ def get_num_units(self) -> int:
         return self.sorting.get_num_units()
 
     ## extensions zone
-
-    def compute(self, input, save=True, **kwargs):
+    def compute(self, input, save=True, extension_params=None, **kwargs):
         """
         Compute one extension or several extensiosn.
         Internally calls compute_one_extension() or compute_several_extensions() depending on the input type.
 
         Parameters
         ----------
         input: str or dict or list
-            If the input is a string then computes one extension with compute_one_extension(extension_name=input, ...)
-            If the input is a dict then compute several extensions with compute_several_extensions(extensions=input)
+            The extesions to compute, which can be passed as:
+
+            * a string: compute one extension. Additional parameters can be passed as key word arguments.
+            * a dict: compute several extensions. The keys are the extension names and the values are dictiopnaries with the extension parameters.
+            * a list: compute several extensions. The list contains the extension names. Additional parameters can be passed with the extension_params
+              argument.
+        save: bool, default: True
+            If True the extension is saved to disk (only of sorting analyzer format is not "memory")
+        extension_params: dict or None, default: None
+            If input is a list, this parameter can be used to specify parameters for each extension.
+            The extension_params keys must be included in the input list.
+        **kwargs:
+            All other kwargs are transmitted to extension.set_params() (if input is a string) or job_kwargs
+
+        Returns
+        -------
+        extension: SortingAnalyzerExtension | None
+            The extension instance if input is a string, None otherwise.
+
+        Examples
+        --------
+        This function accepts the following possible signatures for flexibility:
+
+        Compute one extension, with parameters:
+        >>> analyzer.compute("waveforms", ms_before=1.5, ms_after=2.5)
+
+        Compute two extensions with a list as input and with default parameters:
+        >>> analyzer.compute(["random_spikes", "waveforms"])
+
+        Compute two extensions with dict as input, one dict per extension
+        >>> analyzer.compute({"random_spikes":{}, "waveforms":{"ms_before":1.5, "ms_after", "2.5"}})
+
+        Compute two extensions with an input list specifying custom parameters for one
+        (the other will use default parameters):
+        >>> analyzer.compute(
+            ["random_spikes", "waveforms"],
+            extension_params={"waveforms":{"ms_before":1.5, "ms_after", "2.5"}}
+        )
```

Comment:
This example looks great to me!

I'm sure Chris or I can update the docs themselves soon. (I think we need a section of compute, deleting extensions during recompute) amongst other things. So I think the scope for the docs is a new PR!